assertk/src/jvmTest/kotlin/test/assertk/assertions/JavaAnyTest.kt

```diff
@@ -207,6 +207,22 @@ class JavaAnyTest {
             BasicObject::failing
         )
     }
+
+    @Test
+    fun isEqualToIgnoringGivenProperties_passes_subclass() {
+        fun testObject(): TestObject {
+            return BasicObject(str = "Rarity")
+        }
+
+        assertThat(
+            testObject()
+        ).isEqualToIgnoringGivenProperties(
+            BasicObject(str = "notRarity"),
+            BasicObject::str,
```

Comment:
Prior to the change this would fail to compile, since at compilation time `T == TestObject`, so the method only allows passing properties of `TestObject` and not its subclasses.